fairscale/nn/misc/flatten_params_wrapper.py

```diff
@@ -368,12 +370,70 @@ def _unflatten_params_as_views(self) -> None:
         """Unlike ``_unflatten_params``, this function unflatten into views and keep
         self.flat_param unchanged.
         """
+        logger.info("CHRISLOG: _unflatten_params_as_views() called")
         assert self.is_flattened
-        ps = self.get_param_views()
+        # ps = self.get_param_views()
+
+        """Return a generator of views that map to the original parameters."""
+
+        """Used to get a generator over all views from a list of external data list."""
+        params = self.flat_params
+        external_data_list = [None] * len(params)
+        assert len(external_data_list) == len(
+            params
+        ), f"Incorrect external data list: {len(external_data_list)} vs. {len(params)}"
+
+        # Post accumulation hook so we can call backward() on original leaf params at last microbatch
+        import functools
+
+        def _post_accumulation_hook(new_param_stop_grad, new_param):
+            # TODO: make it only call backward() only for last microbatch (within FSDP no_sync)
```

Comment:
Could you please try to split this into three `logger.info` entries? I think the printing of `new_param` failed as the new_param we used to build partial function was gone...